com.unity.render-pipelines.high-definition/Editor/Material/Nature/HDSpeedTree8MaterialUpgrader.cs

```diff
@@ -0,0 +1,104 @@
+using UnityEngine;
+using UnityEngine.Rendering.HighDefinition;
+
+namespace UnityEditor.Rendering.HighDefinition
+{
+    /// <summary>
+    /// SpeedTree8 material upgrader for HDRP.
+    /// </summary>
+	class HDSpeedTree8MaterialUpgrader : SpeedTree8MaterialUpgrader
+    {
+        /// <summary>
+        /// Creates a SpeedTree8 material upgrader for HDRP.
+        /// </summary>
+        /// <param name="sourceShaderName">Original shader name.</param>
+        /// <param name="destShaderName">Upgrade shader name.</param>
+        public HDSpeedTree8MaterialUpgrader(string sourceShaderName, string destShaderName)
+			: base(sourceShaderName, destShaderName, HDSpeedTree8MaterialFinalizer)
+	{
+        }
+
+        public static void HDSpeedTree8MaterialFinalizer(Material mat)
+        {
+            SetHDSpeedTree8Defaults(mat);
+            SpeedTree8MaterialFinalizer(mat);
+            HDShaderUtils.ResetMaterialKeywords(mat);
+        }
+        /// <summary>
+        /// Checks if a given material is an HD SpeedTree8 material.
+        /// </summary>
+        /// <param name="mat">Material to check.</param>
```

Comment:
Proposed rewrite:
Material to check for compatibility with SpeedTree and HDRP

Reply:
What about:
"Material to check for an HDRP-compatible SpeedTree 8 shader."

Reply:
Looks good to me!